backtracking/minimax.py

```diff
@@ -9,26 +9,54 @@
 """
 
 
-def minimax(Depth, nodeIndex, isMax, scores, height):
-
-    if Depth == height:
-        return scores[nodeIndex]
-
-    if isMax:
+def minimax(depth: int, node_index: int, is_max: bool,
+            scores: List[int], height: float) -> int:
+    """
+    >>> import math
+    >>> scores = [90, 23, 6, 33, 21, 65, 123, 34423]
+    >>> height = math.log(len(scores), 2)
+    >>> minimax(0, 0, True, scores, height)
+    65
+    >>> minimax(-1, 0, True, scores, height)
+    Traceback (most recent call last):
+        ...
+    ValueError: Depth cannot be less than 0
+    >>> minimax(0, 0, True, [], 2)
+    Traceback (most recent call last):
+        ...
+    ValueError: Scores cannot be empty
+    >>> scores = [3, 5, 2, 9, 12, 5, 23, 23]
+    >>> height = math.log(len(scores), 2)
+    >>> minimax(0, 0, True, scores, height)
+    12
+    >>> minimax('1', 2, True, [], 2 )
+    Traceback (most recent call last):
+        ...
+    TypeError: '<' not supported between instances of 'str' and 'int'
+    """
+
+    if depth < 0:
+        raise ValueError("Depth cannot be less than 0")
+
+    if len(scores) == 0:
+        raise ValueError("Scores cannot be empty")
+
+    if depth == height:
+        return scores[node_index]
+
+    if is_max:
         return max(
-            minimax(Depth + 1, nodeIndex * 2, False, scores, height),
-            minimax(Depth + 1, nodeIndex * 2 + 1, False, scores, height),
+            minimax(depth + 1, node_index * 2, False, scores, height),
+            minimax(depth + 1, node_index * 2 + 1, False, scores, height),
         )
+
     return min(
-        minimax(Depth + 1, nodeIndex * 2, True, scores, height),
-        minimax(Depth + 1, nodeIndex * 2 + 1, True, scores, height),
+        minimax(depth + 1, node_index * 2, True, scores, height),
+        minimax(depth + 1, node_index * 2 + 1, True, scores, height),
     )
 
 
 if __name__ == "__main__":
+    import doctest
 
-    scores = [90, 23, 6, 33, 21, 65, 123, 34423]
```

Comment:
These five lines show the user something when they run the program.  Removing them means that the program shows the user nothing when run.

Reply:
@cclauss thank you for your feedback, I have added these lines as method main, look at it if is it right way or not?